pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -1403,3 +1406,38 @@ func TestReadPreNUMACheckpoint(t *testing.T) {
 	err = m.readCheckpoint()
 	require.NoError(t, err)
 }
+
+func TestGetTopologyHintsWithUpdates(t *testing.T) {
```

Comment:
Were you able to test if this test constantly fails without your change?

Reply:
yeah, it reproduces the problem

```
o test -timeout 30s -run ^TestGetTopologyHintsWithUpdates$ k8s.io/kubernetes/pkg/kubelet/cm/devicemanager -race -v
=== RUN   TestGetTopologyHintsWithUpdates
I1228 18:09:37.554520 3807131 fake_topology_manager.go:33] "NewFakeManager"
I1228 18:09:37.554769 3807131 manager.go:455] "Failed to read data from checkpoint" checkpoint="kubelet_internal_checkpoint" err="checkpoint is not found"
I1228 18:09:37.559048 3807131 stub.go:143] "Starting to serve on socket" socket="/tmp/device_plugin158084657/device-plugin.sock"
I1228 18:09:37.559402 3807131 plugin_manager.go:118] "Starting Kubelet Plugin Manager"
I1228 18:09:37.562561 3807131 stub.go:164] "GetInfo"
E1228 18:09:37.562939 3807131 goroutinemap.go:150] Operation for "/tmp/device_plugin158084657/server.sock" failed. No retries permitted until 2022-12-28 18:09:38.062878393 +0000 UTC m=+0.578261683 (durationBeforeRetry 500ms). Error: RegisterPlugin error -- failed to get plugin info using RPC GetInfo at socket /tmp/device_plugin158084657/server.sock, err: rpc error: code = Unimplemented desc = unknown service pluginregistration.Registration
I1228 18:09:37.566228 3807131 stub.go:237] "ListAndWatch"
==================
WARNING: DATA RACE
Write at 0x00c0008210e0 by goroutine 120:
  runtime.mapassign_faststr()
      /usr/local/google/home/aojea/sdk/go1.19.4/src/runtime/map_faststr.go:203 +0x0
  k8s.io/kubernetes/pkg/kubelet/cm/devicemanager.(*ManagerImpl).genericDeviceUpdateCallback()
      /usr/local/google/home/aojea/src/kubernetes/pkg/kubelet/cm/devicemanager/manager.go:250 +0xda
  k8s.io/kubernetes/pkg/kubelet/cm/devicemanager.(*ManagerImpl).genericDeviceUpdateCallback-fm()
      <autogenerated>:1 +0x77
```